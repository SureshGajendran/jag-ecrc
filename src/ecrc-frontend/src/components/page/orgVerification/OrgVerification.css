.eligibleList {
  list-style: disc;
}

.contactList {
  list-style-type: lower-roman;
}

.smallTableHeader {
  font-weight: 400;
}
