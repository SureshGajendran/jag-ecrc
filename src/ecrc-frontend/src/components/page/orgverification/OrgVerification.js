import React, { useEffect } from "react";
import PropTypes from "prop-types";

import "../page.css";
import Header from "../../base/header/Header";
import Footer from "../../base/footer/Footer";
import SideCard from "../../base/sideCard/SideCard";
import { Button } from "../../base/button/Button";

export default function OrgVerification({
  page: {
    pageLayout: { header, sideCard1, sideCard2 },
    org
  }
}) {
  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const orgVerification = () => {
    alert("You verified the org!");
  };

  const button = {
    label: "Verify",
    buttonStyle: "btn btn-primary",
    buttonSize: "btn btn-sm",
    type: "submit"
  };

  return (
    <main>
      <Header header={header} />
      <div className="page">
        <div className="content">
          <Button button={button} onClick={orgVerification} />
        </div>
        <div className="sidecard">
          <SideCard sideCard={sideCard1} />
          <SideCard sideCard={sideCard2} />
        </div>
      </div>
      <Footer />
    </main>
  );
}

OrgVerification.propTypes = {
  page: PropTypes.shape({
    pageLayout: PropTypes.shape({
      header: PropTypes.shape({
        name: PropTypes.string.isRequired
      }),
      sideCard1: PropTypes.shape({
        heading: PropTypes.string.isRequired,
        content: PropTypes.string.isRequired
      }),
      sideCard2: PropTypes.shape({
        heading: PropTypes.string.isRequired,
        content: PropTypes.string.isRequired
      })
    }).isRequired
  }).isRequired
};
