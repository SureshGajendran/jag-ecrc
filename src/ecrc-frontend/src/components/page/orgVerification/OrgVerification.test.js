import React from "react";
import { create } from "react-test-renderer";

import OrgVerification from "./OrgVerification";
import { MemoryRouter } from "react-router-dom";

describe("OrgVerification Component", () => {
  test("Matches the snapshot", () => {
    const header = {
      name: "Criminal Record Check"
    };

    const org = {
      orgNm: "Test Org Name",
      addressLine1: "123 Somewhere Lane",
      cityNm: "Nowhere",
      provinceNm: "British Columbia",
      countryNm: "Canada",
      contactPhoneNo: "250 123 4567",
      orgApplicantRelationship: "Employee"
    };

    const setOrg = () => {};

    const page = {
      org,
      setOrg,
      header
    };

    const orgValidationPage = create(
      <MemoryRouter>
        <OrgVerification page={page} org={org} />
      </MemoryRouter>
    );
    expect(orgValidationPage.toJSON()).toMatchSnapshot();
  });
});
