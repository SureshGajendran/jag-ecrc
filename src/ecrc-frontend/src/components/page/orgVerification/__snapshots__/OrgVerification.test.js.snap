// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgVerification Component Matches the snapshot 1`] = `
<main>
  <header>
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="https://www2.gov.bc.ca"
        >
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-none d-md-block"
            height="44"
            src="/images/bcid-logo-rev-en.svg"
            width="181"
          />
          <img
            alt="B.C. Government Logo"
            className="img-fluid d-md-none"
            height="44"
            src="/images/bcid-symbol-rev.svg"
            width="64"
          />
        </a>
        <div
          className="navbar-brand"
        >
          Criminal Record Check
        </div>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNavAltMarkup"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          <div
            className="navbar-nav"
          >
            <a
              className="nav-item nav-link"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    className="page"
  >
    <div
      className="content col-md-8"
    >
      <h1>
        Organization Information
      </h1>
      <p>
        You have provided the access code of the organization that has requested you to complete a criminal record check. Below are the details of that organization. Please confirm that this is the organization. If itâ€™s not, please do NOT proceed and contact the organization that has requested you to complete a criminal record check.
      </p>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              Organization Information
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Organization Name
            </td>
            <td>
              Test Org Name
            </td>
          </tr>
          <tr>
            <td>
              Address
            </td>
            <td>
              123 Somewhere Lane
Nowhere
British Columbia
undefined
Canada
            </td>
          </tr>
          <tr>
            <td>
              Phone Number
            </td>
            <td>
              250 123 4567
            </td>
          </tr>
        </tbody>
      </table>
      <table>
        <thead>
          <tr>
            <th
              colSpan="2"
            >
              Organization Type
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              Role
            </td>
            <td>
              Employee
            </td>
          </tr>
        </tbody>
      </table>
      <div>
        <p>
          To continue with your online criminal record check request, please ensure you meet the following criteria:
        </p>
        <ul>
          <li>
            You must have a BC Service Card Account. If you do not already have a BC Service Card Account, you can initiate the process at the
            <a
              href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card/login-with-card"
            >
              BC Service Card website
            </a>
            .
          </li>
          <li>
            Pay a fee by credit card (Visa, MasterCard, or AMEX). For volunteers completing a request for criminal record check, no payment is required.
          </li>
        </ul>
        <p>
          Once the criminal record check is complete, the organization noted above will receive the results.
        </p>
        <p>
          By selecting continue, you are consenting to have your information released to this organization.
        </p>
      </div>
      <div
        className="buttons"
      >
        <button
          className="btn btn-primary btn btn-sm"
          onClick={[Function]}
          type="submit"
        >
          Back
        </button>
        <button
          className="btn btn-primary btn btn-sm"
          onClick={[Function]}
          type="submit"
        >
          Continue
        </button>
      </div>
    </div>
    <div
      className="sidecard"
    >
      <div>
        <div
          className="wide-dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="round-icon-wrapper"
                    >
                      <svg
                        className="side-card-icon"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        strokeWidth="0"
                        style={
                          Object {
                            "color": undefined,
                          }
                        }
                        viewBox="0 0 576 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M528 32H48C21.5 32 0 53.5 0 80v16h576V80c0-26.5-21.5-48-48-48zM0 432c0 26.5 21.5 48 48 48h480c26.5 0 48-21.5 48-48V128H0v304zm352-232c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16zm0 64c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16zm0 64c0-4.4 3.6-8 8-8h144c4.4 0 8 3.6 8 8v16c0 4.4-3.6 8-8 8H360c-4.4 0-8-3.6-8-8v-16zM176 192c35.3 0 64 28.7 64 64s-28.7 64-64 64-64-28.7-64-64 28.7-64 64-64zM67.1 396.2C75.5 370.5 99.6 352 128 352h8.2c12.3 5.1 25.7 8 39.8 8s27.6-2.9 39.8-8h8.2c28.4 0 52.5 18.5 60.9 44.2 3.2 9.9-5.2 19.8-15.6 19.8H82.7c-10.4 0-18.8-10-15.6-19.8z"
                        />
                      </svg>
                    </div>
                    <span
                      className="sideCardTitle col-lg-9"
                    >
                      Using the BC Services Card
                    </span>
                  </div>
                </h2>
                <div
                  className="submit-content"
                >
                  <p>
                    <div
                      class="side-card-text"
                    >
                      If you have a BC Services Card, you must use it to log in to the online security verification page.
                    </div>
                    <p>
                      <a
                        href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card"
                      >
                        <div
                          class="side-card-link"
                        >
                          Learn more about the BC Services Card.
                        </div>
                      </a>
                    </p>
                    <div
                      class="side-card-text"
                    >
                      Eligibility for a BC Services Card is the same as for the Medical Services Plan(MSP).
                    </div>
                    <p>
                      <a
                        href="https://www2.gov.bc.ca/gov/content/health/health-drug-coverage/msp/bc-residents/eligibility-and-enrolment/are-you-eligible"
                      >
                        <div
                          class="side-card-link"
                        >
                          Learn more about BC Services Card eligibility.
                        </div>
                      </a>
                    </p>
                  </p>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  Useful Links
                </h2>
                <div
                  className="submit-content"
                >
                  <p>
                    <li
                      className="menuItem"
                    >
                      <a
                        href="/tbd"
                      >
                        I'm an employee or volunteer
                      </a>
                    </li>
                    <li
                      className="menuItem"
                    >
                      <a
                        href="https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check/electronic-identity-verification-eiv"
                      >
                        Electronic Identity Verification (EIV)
                      </a>
                    </li>
                    <li
                      className="menuItem"
                    >
                      <a
                        href="https://www2.gov.bc.ca/gov/content/safety/crime-prevention/criminal-record-check/results-and-reconsiderations"
                      >
                        Results and Reconsideration
                      </a>
                    </li>
                  </p>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="dashboard-spacing"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            className="row"
          >
            <div
              className="col-12"
              style={
                Object {
                  "paddingTop": "30px",
                  "position": "relative",
                }
              }
            >
              <p />
              <section
                className="submit-container"
                id="blue-section"
              >
                <h2
                  className="heading-style"
                >
                  Contact the Criminal Records Review Program
                </h2>
                <div
                  className="submit-content"
                >
                  <p>
                    <div
                      style={
                        Object {
                          "fontSize": "16px",
                        }
                      }
                    >
                      For questions about criminal record checks, contact the Criminal Records Review Program Monday to Friday, 8:30 a.m. - 4:30 p.m.
                    </div>
                    <div
                      style={
                        Object {
                          "paddingTop": "20px",
                        }
                      }
                    >
                      <div>
                        <span
                          className="contact-title"
                        >
                          Fax: 
                        </span>
                         250 356-1889
                      </div>
                      <div>
                        <span
                          className="contact-title"
                        >
                          Office: 
                        </span>
                        Toll free - 1 855 587-0185 (press option 2)
                      </div>
                      <div>
                        <span
                          className="contact-title"
                        >
                          Email: 
                        </span>
                        <a
                          className="email"
                          href="mailto:criminalrecords@gov.bc.ca"
                        >
                          criminalrecords@gov.bc.ca
                        </a>
                      </div>
                    </div>
                  </p>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer"
    style={Object {}}
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark"
    >
      <div
        className="container"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
              rel="noopener noreferrer"
              target="_blank"
            >
              Disclaimer
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/privacy"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/accessibility"
              rel="noopener noreferrer"
              target="_blank"
            >
              Accessibility
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/copyright"
              rel="noopener noreferrer"
              target="_blank"
            >
              Copyright
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="https://www2.gov.bc.ca/gov/content/home/contact-us"
              rel="noopener noreferrer"
              target="_blank"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </footer>
</main>
`;
